adam-core/src/main/scala/org/bdgenomics/adam/rdd/read/AlignmentRecordRDDFunctions.scala

```diff
@@ -419,22 +419,35 @@ class AlignmentRecordRDDFunctions(rdd: RDD[AlignmentRecord])
           record.getReadName.toString.dropRight(2)
       })
 
-    if (validationStringency == ValidationStringency.STRICT) {
-      val readIDsWithCounts: RDD[(String, Int)] = readsByID.mapValues(_.size)
-      val unpairedReadIDsWithCounts: RDD[(String, Int)] = readIDsWithCounts.filter(_._2 != 2)
-      maybePersist(unpairedReadIDsWithCounts)
-
-      val numUnpairedReadIDsWithCounts: Long = unpairedReadIDsWithCounts.count()
-      if (numUnpairedReadIDsWithCounts != 0) {
-        val readNameOccurrencesMap: collection.Map[Int, Long] = unpairedReadIDsWithCounts.map(_._2).countByValue()
-        throw new Exception(
-          "Found %d read names that don't occur exactly twice:\n%s\n\nSamples:\n%s".format(
-            numUnpairedReadIDsWithCounts,
-            readNameOccurrencesMap.map(p => "%dx:\t%d".format(p._1, p._2)).mkString("\t", "\n\t", ""),
-            unpairedReadIDsWithCounts.take(100).map(_._1).mkString("\t", "\n\t", "")
-          )
-        )
-      }
+    validationStringency match {
+      case ValidationStringency.STRICT | ValidationStringency.LENIENT =>
+        val readIDsWithCounts: RDD[(String, Int)] = readsByID.mapValues(_.size)
+        val unpairedReadIDsWithCounts: RDD[(String, Int)] = readIDsWithCounts.filter(_._2 != 2)
+        maybePersist(unpairedReadIDsWithCounts)
+
+        val numUnpairedReadIDsWithCounts: Long = unpairedReadIDsWithCounts.count()
+        if (numUnpairedReadIDsWithCounts != 0) {
+          val readNameOccurrencesMap: collection.Map[Int, Long] = unpairedReadIDsWithCounts.map(_._2).countByValue()
+
+          val msg =
+            List(
+              s"Found $numUnpairedReadIDsWithCounts read names that don't occur exactly twice:",
+
+              readNameOccurrencesMap.map({
+                case (numOccurrences, numReadNames) => s"${numOccurrences}x:\t$numReadNames"
+              }).take(100).mkString("\t", "\n\t", ""),
+              "",
```

Comment:
Nit: if we have more than 100 improperly paired reads, then I would think we should print an ellipsis at the end of the message.